bootcompiler/src/main/scala/org/mozartoz/bootcompiler/fastparse/Parser.scala

```diff
@@ -515,7 +515,7 @@ object Parser {
     P(`proc` ~/ procFlags ~ "{" ~ dollarOrExpr ~/ formalArgs ~ "}" ~/ inPhrase ~ `end`).map {
       case (flags, name, args0, body0) =>
         val (args, body) = postProcessArgsAndBody(args0, body0)
-        val proc = ProcPhrase("", args, body, flags)
+        val proc = ProcPhrase(nameOf(name.getOrElse(null)), args, body, flags)
```

Comment:
I think nameOf should only be called when name is bound, for example by duplicating the code in the `match` below.